pandas/core/strings.py

```diff
@@ -2989,33 +2990,34 @@ def rindex(self, sub, start=0, end=None):
     3              sWaPcAsE
     dtype: object
     """)
-    _shared_docs['lower'] = dict(type='lowercase', method='lower', version='')
-    _shared_docs['upper'] = dict(type='uppercase', method='upper', version='')
-    _shared_docs['title'] = dict(type='titlecase', method='title', version='')
-    _shared_docs['capitalize'] = dict(type='be capitalized',
-                                      method='capitalize', version='')
-    _shared_docs['swapcase'] = dict(type='be swapcased', method='swapcase',
-                                    version='')
-    _shared_docs['casefold'] = dict(type='be casefolded', method='casefold',
-                                    version='\n    .. versionadded:: 0.25.0\n')
+    _doc_args = {}  # type: Dict[str, Dict[str, str]]
+    _doc_args['lower'] = dict(type='lowercase', method='lower', version='')
```

Comment:
The values of `_shared_docs` were either strings or a dict of strings, which would yield the following error:

```sh
pandas/core/strings.py:2992: error: Incompatible types in assignment (expression has type "Dict[str, str]", target has type "str")
```

Seemed logical to create a separate container to keep types well defined